package RemyBuffers;

message WhiskerTree {
  optional MemoryRange domain = 1;

  repeated WhiskerTree children = 2;

  optional Whisker leaf = 3;

  optional ConfigRange config = 4;
}

message MemoryRange {
  optional Memory lower = 11;
  optional Memory upper = 12;
}

message Memory {
  optional double rec_send_ewma = 21;
  optional double rec_rec_ewma = 22;
  optional double rtt_ratio = 23;
}

message Whisker {
  optional sint32 window_increment = 31;
  optional double window_multiple = 32;
  optional double intersend = 33;

  optional MemoryRange domain = 34;
}

message ConfigRange {
  required double bw_low = 41;
  required double bw_high = 42;
  required double rtt_low = 43;
  required double rtt_high = 44;
  required uint32 min_senders = 45;
  required uint32 max_senders = 46;
  required double mean_off_duration = 47;
  required double mean_on_duration = 48;
}
